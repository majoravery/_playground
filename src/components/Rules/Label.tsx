import React from "react";
import { TextInput } from "@deliveryhero/armor";

interface Props {
  value: string;
  width?: number;
}

const Label: React.FC<Props> = ({ value, width }) => {
  return (
    <div className="rule-label__container" style={{ width: width || "auto" }}>
      {value}
    </div>
  );
};

export default Label;
