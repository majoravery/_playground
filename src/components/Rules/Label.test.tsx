import { render } from '@testing-library/react';
import Label from './Label';

describe('Label', () => {
  it('should render and match snapshot', () => {
    const { container, rerender } = render(<Label value='Test Value' />);
    expect(container).toMatchSnapshot();
    rerender(<Label value='Test Value' width={200} />);
    expect(container).toMatchSnapshot();
  });
});
