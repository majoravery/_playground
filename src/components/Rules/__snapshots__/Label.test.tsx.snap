// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render and match snapshot 1`] = `
<div>
  <div
    class="rule-label__container"
    style="width: auto;"
  >
    Test Value
  </div>
</div>
`;

exports[`Label should render and match snapshot 2`] = `
<div>
  <div
    class="rule-label__container"
    style="width: 200px;"
  >
    Test Value
  </div>
</div>
`;
